sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -662,7 +661,62 @@ Exceptions:
 
 |===
 
-Table 8. Member functions of the `command_graph` class (executable graph update).
+Table 8. Member functions of the `command_graph` class for queue recording.
+[cols="2a,a"]
+|===
+|Member function|Description
+
+|
+[source, c++]
+----
+using namespace ext::oneapi::experimental;
+bool begin_recording(const std::vector<queue>& queues)
+----
+
+|Synchronously changes the state of each queue in `queues` to the
+`queue_state::recording` state.
+
+Parameters:
+
+* `queues` - List of `sycl::queue` objects to change to the
+  `queue_state::recording` state and start recording commands to the graph
+  instance.
+
+Returns: `true` if any queue in `queues` has its state changed from
+`queue_state::executing` to `queue_state::recording`, `false` otherwise.
```

Comment:
Thinking about this some more, I'm leaning towards have both overloads for `begin_recording`, but it's not so much about the return info usefulness - I think that'll be mostly looked at during application development/debugging where the extra cost of the vector creation for a single queue isn't an issue if you were iterating to debug multiple queues.

Instead, I think that the single queue use case will be the most common scenario, so having the vector creation overhead in the most common usage seems undesirable. But equally for multiple queue usages, iteratively calling the entry-point is also undesirable. So think having both entry-points for recording seems fine.

For `end_recording()`, I'm less sure if we need overloads since I expect the empty vector to be the most common usage.

Reply:
That makes sense to me!